File: extension/core/src/test/java/io/zeebe/camel/ZeebeComponentTest.java
package io.zeebe.camel;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;

import io.zeebe.camel.handler.task.TaskEndpoint;
import io.zeebe.camel.handler.task.TaskUri;
import io.zeebe.camel.handler.universal.UniversalEventEndpoint;
import io.zeebe.camel.handler.universal.UniversalEventUri;
import io.zeebe.client.ZeebeClient;
import org.apache.camel.impl.DefaultCamelContext;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.mockito.Mockito;

public class ZeebeComponentTest
{

    private final ZeebeClient client = mock(ZeebeClient.class);
    private final ZeebeComponent component = new ZeebeComponent(client);
    private DefaultCamelContext camelContext;

    @Before
    public void setUp() throws Exception
    {
        camelContext = new DefaultCamelContext();
        camelContext.addComponent(ZeebeComponent.SCHEME, component);
    }

    @Test
    public void create_universalEventHandler() throws Exception
    {
        final String uri = UniversalEventUri.topic(ZeebeComponent.DEFAULT_TOPIC).name("test").get();
        UniversalEventEndpoint endpoint = (UniversalEventEndpoint) component.createEndpoint(uri);

        assertThat(endpoint).isNotNull();

        assertThat(endpoint.getTopic()).isEqualTo(ZeebeComponent.DEFAULT_TOPIC);
        assertThat(endpoint.getName()).isEqualTo("test");
    }


    @Test
    public void create_taskHandler() throws Exception
    {
        final String uri = TaskUri.topic(ZeebeComponent.DEFAULT_TOPIC).type("type").lockOwner("lockedby").get();
        TaskEndpoint endpoint = (TaskEndpoint) component.createEndpoint(uri);

        assertThat(endpoint).isNotNull();
        assertThat(endpoint.getTopic()).isEqualTo(ZeebeComponent.DEFAULT_TOPIC);
        assertThat(endpoint.getOwner()).isEqualTo("lockedby");
        assertThat(endpoint.getType()).isEqualTo("type");
    }



    @After
    public void tearDown() throws Exception
    {
        camelContext.stop();
    }
}

File: extension/core/src/test/kotlin/ZeebeComponentTest.kt
package io.zeebe.camel

import io.zeebe.camel.endpoint.SubscribeJobWorkerEndpoint
import io.zeebe.client.ZeebeClient
import org.apache.camel.impl.DefaultCamelContext
import org.assertj.core.api.Assertions.assertThat
import org.junit.After
import org.junit.Test
import org.mockito.Answers
import org.mockito.Mockito

@Deprecated("seems like this is not needed anymore?")
class ZeebeComponentTest {

    private val camel = DefaultCamelContext()
    private val client = Mockito.mock(ZeebeClient::class.java, Answers.RETURNS_DEEP_STUBS)

    init {
        camel.addComponent(ZeebeComponent.SCHEME, ZeebeComponent(client))
    }

    @Test
    fun `create jobworker endpoint`() {

        val endpoint = camel.getEndpoint("zeebe:jobworker", SubscribeJobWorkerEndpoint::class.java)

    }

    @After
    fun tearDown() {
        camel.stop()
    }
}
